src/styles/calendar/calendar.scss

```diff
@@ -0,0 +1,48 @@
+@use '../utils/variables' as *;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  padding: 10px;
+  width: $calendar-width;
+  gap: $gap;
+  align-content: flex-end;
+
+  @for $i from 1 through 31 {
+    &--month-length-#{$i} div:nth-child(-n + #{$i}) {
+      display: flex;
+    }
+  }
+
+  @each $day, $position in $start-day {
+    &--start-day-#{$day} div:first-child {
+      margin-left: $position;
+    }
+  }
+
+  &__day {
+    display: none;
+    justify-content: center;
+    align-items: center;
+    background-color: $day-background-color;
+    width: $day-width;
+    height: $day-height;
+    border: 1px solid black;
+    box-sizing: border-box;
+    transition: 0.5s;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: 30px;
+      }
+    }
+
+    &:hover {
+      transform: translateY(-20px);
+      cursor: pointer;
+      background-color: #ffbfcb;
```

Comment:
you already have variable for day element color so to be consistent you should also create variable for background-color on hover